sdks/python/apache_beam/runners/interactive/caching/streambasedcache.py

```diff
@@ -0,0 +1,491 @@
+          subscription_path, self.topic.name)
+    except gexc.AlreadyExists:
+      if mode == "error":
+        raise IOError(
+            "Subscription '{}' already exists.".format(subscription_path))
+      elif mode == "overwrite":
+        sub_client.delete_subscription(subscription_path)
+        self.subscription = sub_client.create_subscription(
+            subscription_path, self.topic.name)
+      else:
+        self.subscription = sub_client.get_subscription(subscription_path)
+
+    snapshot_path = sub_client.snapshot_path(self.project, self.topic_name)
+    try:
+      self.snapshot = sub_client.create_snapshot(snapshot_path,
+                                                 self.subscription.name)
+    except gexc.AlreadyExists:
+      if mode == "error":
+        raise IOError("Snapshot '{}' already exists.".format(snapshot_path))
+      elif mode == "overwrite":
+        sub_client.delete_snapshot(snapshot_path)
+        self.snapshot = sub_client.create_snapshot(snapshot_path,
+                                                   self.subscription.name)
+      else:
+        # See: https://github.com/googleapis/google-cloud-python/issues/8554
+        self.snapshot = pubsub.types.Snapshot()
+        self.snapshot.name = snapshot_path
+    except gexc.MethodNotImplemented:
+      self.snapshot = None
+
+    self._child_subscriptions.append(self.subscription)
+    self._init_finalizers()
+
+  @property
+  def timestamp(self):
+    return self._timestamp
+
+  @property
+  def persist(self):
+    return self._persist
+
+  @persist.setter
+  def persist(self, persist):
+    self._persist = persist
+    self._init_finalizers()
+
+  def reader(self, seek_to_start=True, **kwargs):
+    self._assert_topic_exists()
+
+    reader_kwargs = self._reader_kwargs.copy()
+    reader_kwargs.update(kwargs)
+
+    if "subscription" not in reader_kwargs:
+      reader_kwargs["subscription"] = self._create_child_subscription(
+          seek_to_start=seek_to_start).name
+
+    reader = PatchedPubSubReader(self, self._reader_class, **reader_kwargs)
+    return reader
+
+  def writer(self):
+    self._timestamp = time.time()
+    writer = PatchedPubSubWriter(self, self._writer_class, (self.location,),
+                                 self._writer_kwargs)
+    return writer
+
+  @contextlib.contextmanager
+  def read_to_queue(self,
+                    seek_to_start=True,
+                    **kwargs):
+    self._assert_topic_exists()
+
+    reader_kwargs = self._reader_kwargs.copy()
+    reader_kwargs.update(kwargs)
+
+    if "subscription" in reader_kwargs:
+      created_subsciption = None
+    else:
+      created_subsciption = self._create_child_subscription(
+          seek_to_start=seek_to_start)
+      reader_kwargs["subscription"] = created_subsciption.name
+      assert created_subsciption in self._child_subscriptions
+
+    @functools.total_ordering
+    class PrioritizedTimestampedValue(TimestampedValue):
```

Comment:
`TimestampedValue` does implement comparisons, but those comparisons are done primarily based on the value rather than the timestamp. Here, we want the values with the oldest timestamp to come first. Would you think that changing the name of this class would make the intention more clear?